### ISSUE TITLE
<svelte:head> only one time such as a script context="module" do for script.

### ISSUE THREAD

**Issue by giuseppe-palmeri** [sveltejs/kit#10502]
### Describe the problem

I would like to write a component in the /src/lib folder which, upon its initialization, loads an external js script such as:
```
<script defer async src="https://maps.googleapis.com/maps/api/js?key=YOUR_API_KEY&callback=initMap"></script>
```

To do this I used <svelte:head> like this:
```
<svelte:head>
	<script defer async
	src="https://maps.googleapis.com/maps/api/js?key=YOUR_API_KEY&callback=initMap">
	</script>
</svelte:head>
```

This solution seems to be pretty clean but it is not.
What I expect is that if I place several components of this type on the same page, the information contained in <svelte:head> will be present in the page header only once and not N times as there are instances of the component on the page, as unfortunately happens.

For example this is com.svelte:
```
<script></script>
<svelte:head>
	<script defer async
	src="https://maps.googleapis.com/maps/api/js?key=YOUR_API_KEY&callback=initMap">
	</script>
</svelte:head>
<style><style>
```

and this is my +page.svelte:
```
<Com></Com>
<Com></Com>
<Com></Com>
```
The <svelte:head> will be duplicated 3 times into the header of the page.
But I would like this to happen only once, exactly like it happens with running javascript in a module context.

### Describe the proposed solution

If there is no reason for the described thing to occur then this should be treated as a bug.

Otherwise, something like <svelte:head context="module"> or <svelte:head singlerun=true> should be considered.
So you have control over the behavior of <svelte:head>
Thank you.

### Importance

would make my life easier

**Comment by dummdidumm** [comment]
`<svelte:head>` in its current form shouldn't deduplicate things by default. You can achieve what you want without any new APIs, by using `<script context="module">` to check if the thing was already rendered:

```svelte
<script context="module">
  let inserted = false;
</script>

<script>
  let shouldInsert = !inserted;
  inserted = true;
</script>

<svelte:head>
  {#if shouldInsert}
    <script ...></script>
  {/if}
</svelte:head>
```

**Comment by frudman** [comment]
so the solution proposed above doesn't work in all cases.

In my case, I used the concept above (singleton) to include a <link rel=stylesheet ...css> to the head once, when component is first used, but then not re-included again. Unfortunately, instead of leaving that head entry there (and not re-including it again and again), Svelte (seems to) removes the link from the head altogether, which of course breaks my code that needs that stylesheet present.

I don't have a workaround so far, other than just let it include that link many many times (namely each time the component is used). In my favor, other than the futility of doing that work, I believe that since the <link> file itself is cached, at least the browser doesn't go back to inter-tubes to get that file over and over again.

It would be nice to have as original poster suggested: a module-like approach to placing things in the head (or body). Once, as singletons.
